app/controllers/application_controller.rb

```diff
@@ -20,6 +20,25 @@ class ApplicationController < ActionController::Base
 
   layout 'application'
 
+  # A common method for controllers to find a Dor object.  It accepts any valid
+  # PID (druid:aa111aa1111) or DRUID (aa111aa1111).  DruidTools::Druid parses
+  # input and returns a uniform PID, which is input to Dor.find to obtain a Dor
+  # object.
+  # @see https://github.com/sul-dlss/druid-tools
+  # @param druid [String]
+  # @return object [ActiveFedora::Base] any Dor object
+  # @raises ArgumentError
+  def find_druid(druid)
+    druid = DruidTools::Druid.new(druid).druid
+    Dor.find(druid)
+  rescue ArgumentError
+    flash[:error] = "Invalid DRUID: #{druid}"
+    render status: 400, text: flash[:error] and return
+  rescue ActiveFedora::ObjectNotFoundError # => e
+    flash[:error] = 'Object does not exist in Fedora.'
+    render status: 404, text: flash[:error] and return
```

Comment:
I suggest we merge this, as is, and create a new issue/PR to resolve this debate.  The existing specs required flash messages and I don't know why and I don't think this PR is the place to change that.  So I think the current code satisfies the current specs and your suggestions would break them.  That is not to say that an alternative is better or worse, but simply that it's out of scope here and the current code was crafted to satisfy current specs with robust and reliable behavior.